unison-cli/src/Unison/Codebase/TranscriptParser.hs

```diff
@@ -428,6 +444,13 @@ run verbosity dir stanzas codebase runtime sbRuntime nRuntime config ucmVersion
 
       writeSourceFile :: ScratchFileName -> Text -> IO ()
       writeSourceFile fp contents = do
+        shouldShowSourceChanges <- (== Shown) <$> readIORef hidden
+        when shouldShowSourceChanges $ do
+          atomically (Q.enqueue ucmScratchFileUpdatesQueue (fp, contents))
+        updateVirtualFile fp contents
+
+      updateVirtualFile :: ScratchFileName -> Text -> IO ()
+      updateVirtualFile fp contents = do
         liftIO (modifyIORef' unisonFiles (Map.insert fp contents))
```

Comment:
Does `shouldShowSourceChanges` need to be moved too?
Is that for detecting whether it was triggered by a `unison:hide` block, but those cases call `updateVirtualFile` directly?

Reply:
As written does this hide the UCM-generated scratch output in a `ucm:hide` block? We might want that option, and it's fine that it does, until we figure out what we want. I'm just having trouble telling whether it does the expected thing in various cases.

Reply:
Currently as it's configured, `unison:added-by-ucm` ARE hidden by `ucm:hide`. That's what this `shouldShowSourceChanges` stuff is doing, it's easy to change in the future if we like.

> Does shouldShowSourceChanges need to be moved too?

Nope, I'm separating these functions specifically because `shouldShowSourceChanges` and the ucmScratchFileUpdates only apply to updates caused by UCM.